src/Hash.js

```diff
@@ -90,6 +99,19 @@ function hashString(string) {
   return smi(hashed);
 }
 
+function hashSymbol(sym) {
+  let hashed = symbolMap[sym];
+  if (hashed !== undefined) {
+    return hashed;
+  }
+
+  hashed = nextHash();
+
+  symbolMap[sym] = hashed;
```

Comment:
Note that this is technically a memory leak. I'm just hoping there's no reason that programmers would create an unbounded number of symbols.